runtime/src/system_instruction_processor.rs

```diff
@@ -91,7 +97,12 @@ fn finish_create_account(
         return Err(SystemError::InvalidAccountId.into());
     }
 
-    if lamports > from.account.lamports {
+    let min_balance = match account_is_system(&from.account) {
```

Comment:
I think it'd be better to instead say "no transfers out of nonces, use a withdraw instruction instead"

Reply:
'cause I should've stopped working on this about an hour before I did 😉 

It is of course much cleaner to just deny nonce accounts as `from`. Will update

Reply:
https://github.com/solana-labs/solana/pull/7645/commits/41e9ad3753c6ba1aa4acc576593051b10d13295e